tests/test_box_Box.py

```diff
@@ -439,6 +439,67 @@ def test_plot_2d(self):
         box = freud.box.Box(2, 3, 0, 1, 0, 0, is2D=True)
         box.plot()
 
+    def test_compute_distances_2d(self):
+        box = freud.box.Box(2, 3, 0, 1, 0, 0, is2D=True)
+        points = np.array([[0, 0, 0], [-2.2, -1.3, 0]])
+        query_points = np.array(
+            [[-0.5, -1.3, 0.], [0.5, 0, 0], [-2.2, -1.3, 0.], [0, 0.4, 0]])
+        point_indices = np.array([1, 0, 1, 0])
+        query_point_indices = np.array([0, 1, 2, 3])
+        distances = box.compute_distances(
+            query_points, points, query_point_indices, point_indices)
+        npt.assert_allclose(distances, [0.3, 0.5, 0.0, 0.4], rtol=1e-6)
+
+        # # 1 dimensional array
+        distances = box.compute_distances(
+            query_points, points,
+            query_point_indices[0], point_indices[0])
+        npt.assert_allclose(distances, 0.3, rtol=1e-6)
+
+        with self.assertRaises(IndexError):
+            box.compute_distances(
+                query_points, points, query_point_indices, point_indices + 1)
+        with self.assertRaises(IndexError):
+            box.compute_distances(
+                query_points, points, query_point_indices + 1, point_indices)
+        with self.assertRaises(RuntimeError):
+            box.compute_distances(
+                query_points, points, query_point_indices, point_indices[:-1])
```

Comment:
Is this raising because the length of `query_point_indices` doesn't match the length of `point_indices`? If so, that should be a `ValueError`, not a `RuntimeError`. You might need to change the C++ exception that is thrown so that the Python exception corresponds to that.